File: src/Application Specific/Classes/Entity Collections/InvoiceToBeRaised/InvoiceToBeRaisedValidator.vb
Namespace Entity


Namespace InvoiceToBeRaiseds

Public Class InvoiceToBeRaisedValidator
    'make sure that contact object is valid
    Inherits BaseValidator

    Public Sub Validate(ByVal oInvoiceToBeRaised As InvoiceToBeRaised)

        'make sure that contact object is valid
        If oInvoiceToBeRaised.Errors.Count > 0 Then
            Dim de As DictionaryEntry
            For Each de In oInvoiceToBeRaised.Errors
                Me.AddCriticalMessage(CStr(de.Value))
            Next
        End If

        
	
        If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
            Throw New ValidationException(Me)
        End If

    End Sub

End Class

End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/InvoiceToBeRaised/InvoiceToBeRaisedValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace InvoiceToBeRaiseds
    {
        // make sure that contact object is valid
        public class InvoiceToBeRaisedValidator : BaseValidator
        {
            public void Validate(InvoiceToBeRaised oInvoiceToBeRaised)
            {
                // make sure that contact object is valid
                if (oInvoiceToBeRaised.Errors.Count > 0)
                {
                    foreach (DictionaryEntry de in oInvoiceToBeRaised.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}